src/main/java/io/cdap/plugin/http/sink/batch/HTTPRecordWriter.java

```diff
@@ -86,9 +114,39 @@ public void write(StructuredRecord input, StructuredRecord unused) throws IOExce
       }
       messages.append(message).append(config.getDelimiterForMessages());
     }
+    if (config.getMethod().equals("PUT") || config.getMethod().equals("DELETE") && !placeHolderList.isEmpty()) {
+      configURL = updateURLWithPlaceholderValue(input);
+    }
     StringTokenizer tokens = new StringTokenizer(messages.toString().trim(), config.getDelimiterForMessages());
+
+    // response being null, means it's the first request
+    Long delay = (response == null || config.getReadTimeout() == null) ? 0L : config.getReadTimeout();
+
```

Comment:
variable `response` is never set, delay will always be 0.
Can we use `long` instead ?

Reply:
> variable `response` is never set, delay will always be 0. 
Removed response. 4e4ee6211be03c9e368a811866c8d96ec49518d3

> Can we use `long` instead ?
Changed to primitive type. 4e4ee6211be03c9e368a811866c8d96ec49518d3